codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointResolverInterceptorSpec.java

```diff
@@ -317,16 +318,14 @@ private MethodSpec getCredentialScopeIfPresent() {
                                          .addParameter(ExecutionAttributes.class, "executionAttributes")
                                          .returns(Region.class);
 
-        b.addStatement("$T<?> authScheme = executionAttributes.getAttribute($T.SELECTED_AUTH_SCHEME)",
-                       SelectedAuthScheme.class, SdkInternalExecutionAttribute.class);
-
-        b.addStatement("$T identity = $T.joinLikeSync(authScheme.identity())", Identity.class, CompletableFutureUtils.class);
-        b.beginControlFlow("if (identity instanceof $T)", AwsCredentialsIdentity.class);
-        b.addStatement("$T awsCredentialsIdentity = ($T) identity", AwsCredentialsIdentity.class, AwsCredentialsIdentity.class);
-        b.addStatement("return awsCredentialsIdentity.credentialScope().map($T::of).orElse(null)", Region.class);
-        b.endControlFlow();
+        b.addStatement("$T identityProviders = executionAttributes.getAttribute($T.IDENTITY_PROVIDERS)",
+                       IdentityProviders.class, SdkInternalExecutionAttribute.class);
 
-        b.addStatement("return null");
+        b.addStatement("$T awsCredentialsIdentity = $T.joinLikeSync(identityProviders.identityProvider($T.class)"
+                       + ".resolveIdentity())",
```

Comment:
Is there a way that the call to `identityProviders.identityProvider(...)` can return null or are we sure that is always returns a valid instance?

Reply:
It can be null. We should definitely guard against null.

Reply:
good point it would return null for TokenIdentity, adding null check